content/200-orm/200-prisma-client/150-writing-custom-sql/index.mdx

```diff
@@ -0,0 +1,72 @@
+
+This may happen for several reasons, e.g., because you need to optimize the performance of a specific query or because your data requirements can't be expressed by Prisma Client's query API.
+
+In these cases, you can use one of Prisma Client's advanced querying features via `$queryRaw` or `$executeRaw`, to send raw queries to the database.
+
+In most cases, [TypedSQL](#writing-type-safe-queries-with-prisma-client-and-typedsql) will allow you to express your query in SQL while still leveraging Prisma Client's excellent user experience. However, since TypedSQL is statically typed, there are some cases (such as dynamically generated `WHERE` clauses) that it can't represent. For these cases, you will need to use [`$queryRaw`](/orm/prisma-client/writing-custom-sql/raw-queries#queryraw) or [`$executeRaw`](/orm/prisma-client/writing-custom-sql/raw-queries#executeraw) or their unsafe counterparts.
+
+## Writing type-safe queries with Prisma Client and TypedSQL
+
+:::info
+
+TypedSQL is available in Prisma ORM 5.18.0 and later. For raw database access in previous versions, see [our raw queries documentation](/orm/prisma-client/writing-custom-sql/raw-queries).
+
+:::
+
+### What is TypedSQL?
+
+TypedSQL is new feature in Prisma ORM that bridges the gap between raw SQL and type-safe TypeScript code. It allows you to write SQL queries in dedicated `.sql` files while leveraging the full type safety and developer experience of Prisma Client.
+
+With TypedSQL, you can:
+
+1. Write complex SQL queries using familiar syntax
+2. Benefit from full IDE support and syntax highlighting for SQL
+3. Import your SQL queries as fully typed functions in your TypeScript code
+4. Maintain the flexibility of raw SQL with the safety of Prisma's type system
+
+TypedSQL is particularly useful for:
+
+- Complex reporting queries that are difficult to express using Prisma's query API
+- Performance-critical operations that require fine-tuned SQL
+- Leveraging database-specific features not yet supported in Prisma's API
+
+By using TypedSQL, you can write efficient, type-safe database queries without sacrificing the power and flexibility of raw SQL. This feature allows you to seamlessly integrate custom SQL queries into your Prisma-powered applications, ensuring type safety and improving developer productivity.
+
+For a detailed guide on how to get started with TypedSQL, including setup instructions and usage examples, please refer to our [TypedSQL documentation](/orm/prisma-client/writing-custom-sql/typedsql).
+
+## Raw SQL queries
+
+Prior to version 5.18.0, Prisma Client only supported raw SQL queries that were not type-safe and required manual mapping of the query result to the desired type. While not as ergonomic as [TypedSQL](#writing-type-safe-queries-with-prisma-client-and-typedsql), these queries are still supported and are useful when a query is dynamically generated. 
+
+### Legacy raw queries in relational databases
```

Comment:
As TypedSQL doesn't have an alternative to perform dynamic raw queries, and we have to suggest users to use these queries, how about we have a different name until a time where we don't have to depend on it:

```suggestion
### Alternative approaches to perform raw queries in relational databases
```